proposals/proposals/ 0285-ease-pound-file-transition.md-ease-pound-file-transition.md

```diff
@@ -0,0 +1,162 @@
+
+### Magic identifier default argument mismatch warnings
+
+Swift will not warn when:
+
+* A parameter with a default argument of `#file` is passed to one with a default argument of `#fileID`.
+* A parameter with a default argument of `#fileID` is passed to one with a default argument of `#file`.
+
+Additionally, Swift 5 mode and earlier will not warn when:
+
+* A parameter with a default argument of `#file` is passed to one with a default argument of `#filePath`.
+* A parameter with a default argument of `#filePath` is passed to one with a default argument of `#file`.
+
+In Swift 5 mode, substituting `#file` for `#fileID` *will* actually result in different behavior, so ideally we would warn about this. However, doing that would cause new warnings in existing functions which wrap functions that adopt `#fileID`. In practice, most `#fileID` adopters will work fine when they're passed `#file`—they'll just generate unnecessarily large file strings. 
+
+### Schedule
+
+Ideally, these changes will be included in Swift 5.3, with the "future language mode" parts tied to a frontend flag for testing, rather than a language version mode.
+
+## Source compatibility
+
+Significantly improved compared to SE-0274. Code that uses `#filePath` or `#fileID` will not be compilable in Swift 5.2 and earlier, but such code can continue to use `#file` until a future language version mode permits breaking changes.
+
+## Effect on ABI stability
+
+See SE-0274.
+
+## Effect on API resilience
+
+See SE-0274.
+
+## Alternatives considered
+
+The SE-0274 design was considered and accepted, but did not hold up to real-world use.
+
+In future language version modes, we could deprecate `#file` instead of `#fileID`, creating a situation where users must choose whether to update their uses of `#file` to either `#fileID` or `#filePath`. This would prompt future users to make a choice of behavior, but we don't want to force those users to make an explicit choice; we continue to believe that the concise file string (i.e. `#fileID` string) is the right choice for most uses.
+
+Similarly, we could simply introduce `#fileID` and change the standard library to use it, without adding `#filePath` or changing `#file`'s behavior. But since we think the `#fileID` string is the right default, this would leave us with a design that encouraged the wrong defaults.
+
+We could remove the `#fileID` syntax from this proposal, probably retaining it underscored form as an implementation detail used to transition the standard library asssertions to the new strings in Swift 5 mode. This would give us a cleaner "Swift 6 mode" without a deprecated `#fileID` hanging around, but we don't want to keep potential adopters outside the standard library from gaining the advantages of the new strings if they want them.
```

Comment:
```suggestion
We could remove the `#fileID` syntax from this proposal, probably retaining it in underscored form as an implementation detail used to transition the standard library assertions to the new strings in Swift 5 mode. This would give us a cleaner "Swift 6 mode" without a deprecated `#fileID` hanging around, but we don't want to keep potential adopters outside the standard library from gaining the advantages of the new strings if they want them.
```